### PR TITLE
feat(notes): [INFRA-293] Get notes on highlights

### PR DESCRIPTION
**Pull request by kschelonka**
## Goal
Resolve notes attached to highlights. A highlight can contain a single note, which is stored in DynamoDB.

- Create NotesDataService to retrieve notes from DynamoDB
- Update localstack infrastructure for local integration testing (env vars, etc.)
- Simplify some configuration thanks to DynamoDB Document Client
- Update some fields in the Notes schema in DynamoDB based on reserved words/characters
- Implement resolver with integration tests

Paired with @anthony-liddle 

## I'd love feedback/perspectives on:
- do I need to destroy the dynamo db client somehow when the request is complete? e.g. add it onto the context and use a lifecycle hook plugin? or is it garbage-collected?

## Implementation Decisions
- Using the DynamoDB Document Client instead of the base DynamoDB Client

## References

JIRA ticket:
* [INFRA-293]

[INFRA-293]: https://getpocket.atlassian.net/browse/INFRA-293?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .aws/src/config/index.ts [hunk 1]
```diff
@@ -54,16 +53,10 @@ export const config = {
   },
   dynamodb: {
     notesTable: {
-      key: {
-        name: 'highlightId',
-        type: 'S'
-      },
+      key: 'highlightId',
```
Comment (kschelonka) [inline .aws/src/config/index.ts:56]: Simplifying this since we're using the Document client

File: tsconfig.json [hunk 2]
```diff
@@ -8,7 +8,8 @@
       "es2019",
       "es2020.bigint",
       "es2020.string",
-      "es2020.symbol.wellknown"
+      "es2020.symbol.wellknown",
+      "dom"
```
Comment (kschelonka) [inline tsconfig.json:12]: This fixes a types issue with the DynamoDB library

File: .circleci/config.yml [hunk 3]
```diff
@@ -160,7 +172,7 @@ jobs:
           paths:
             - ./node_modules
       - run:
-          # TODO: Update 
+          # TODO: Update
```
Comment (kschelonka) [inline .circleci/config.yml:175]:
```suggestion
```

File: .vscode/settings.json [hunk 4]
```diff
@@ -1,10 +1,13 @@
 {
-    "editor.codeActionsOnSave": {
+  "editor.defaultFormatter": "esbenp.prettier-vscode",
```
Comment (kkyeboah) [inline .vscode/settings.json:2]: 😤

File: src/resolvers.ts [hunk 5]
```diff
@@ -20,6 +22,15 @@ export const resolvers = {
       return { highlights };
     },
   },
+  Highlight: {
+    note: async (
```
Comment (kkyeboah) [inline src/resolvers.ts:26]: I smell a data loader in the near future

File: src/test/query/notes.integration.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    const variables = { itemId: 1 };
+    const res = await server.executeOperation({
+      query: GET_NOTES,
+      variables,
+    });
+
+    const expectedHighlightWithNote = {
+      id: '1',
+      note: {
+        _createdAt: Math.round(now.getTime() / 1000),
```
Comment (kkyeboah) [inline src/test/query/notes.integration.ts:50]: seen this a couple of times now, let's make it a utility function the next time we see it.

**Review by kkyeboah — APPROVED** [review]
Looks good to me

File: src/resolvers.ts (hunk 5 above)
Comment (anthony-liddle) [reply src/resolvers.ts:26]: smell

Comment (kschelonka) [reply src/resolvers.ts:26]: yep

File: src/test/query/notes.integration.ts (hunk 6 above)
Comment (kschelonka) [reply src/test/query/notes.integration.ts:50]: why it already doesn't exist... idk

**Pull request merged**